Automattic/wp-calypso
# Domains: Add transfer connected domain nudge component

### Changes proposed in this Pull Request

This is a companion PR to diff D73307-code.

As suggested in p1641308839078500-slack-C018R0DDFKJ, we would like to show a nudge to customers who have connected domains that will expire soon. That nudge will encourage customers to transfer their domains to WPCOM, making their domains management easier and all in one place, and increasing our domains under management count.

This PR creates a new `TransferConnectedDomainNudge` component that will show that nudge for domains which have the `showTransferConnectedDomainNudge` property set as `true` - these are connected domains which will expire in 45 days or less. The `transferring it` link in the notice takes the user to the start of the `transfer a domain I own` flow for the domain in question. The nudge will be shown in the domains list and domain settings pages.

#### Screenshots

- Domain list:

<img width="1062" alt="Screen Shot 2022-01-24 at 15 12 52" src="https://user-images.githubusercontent.com/5324818/150840263-33bce349-a7d3-471e-a35e-ee2aed8f2cdd.png">

- Domain settings:

<img width="1048" alt="Screen Shot 2022-01-24 at 15 13 01" src="https://user-images.githubusercontent.com/5324818/150840274-eb296636-903c-4a08-a876-0fb0154705be.png">

### Testing instructions

- This should be tested alongside D73307-code or by using the mocked domains as detailed in the testing intructions in D70509-code
- Sandbox `public-api`
- Build this branch locally or open the live Calypso link
- Select a site where you have a connected domain that's going to expire in 45 or less days
- Go to the domains list and domain management pages
- Ensure the transfer nudge is shown and looks like the screenshots above
- Ensure the "transferring it" link takes you to the start of the "transfer a domain I own" setup.
